test6.go

```diff
@@ -1,61 +1,56 @@
 package main
 
 import (
-    "fmt"
-    "io/ioutil"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
 	"net/http"
 	"os"
-	"log"
-	"io"
-	"encoding/json"
-	"encoding/csv"
 )
 
-type Sheet struct{
-	Sitename string `json:"sitename"`
-	Location string `json:"location"`
-	Address *Address `json:address`
-
+type Sheet struct {
+	Sitename string   `json:"sitename"`
+	Location string   `json:"location"`
+	Address  *Address `json:address`
 }
 
 type Address struct {
 	Number string `json:"number"`
 }
 
-
-
 func main() {
 
 	csvFile, _ := os.Open("test.csv")
-	reader:= csv.NewReader(csvFile)
-	var sheet [] Sheet
+	reader := csv.NewReader(csvFile)
+	var sheet []Sheet
 	for {
-	 line, error:= reader.Read()
-		if error==io.EOF {
+		line, error := reader.Read()
+		if error == io.EOF {
 			break
-		} else if error!= nil{
+		} else if error != nil {
 			log.Fatal(error)
 		}
-		response, err:= http.Get("http://ip-api.com/json/www.google.com?fields=country")
+		response, err := http.Get("http://ip-api.com/json/www.google.com?fields=country")
 		/*I  want to able to pass line[0] which is the website in my test csv in the url*
 		Something like http.Get("http://ip-api.com/json/{line[0}?fields=country") only returns empty JSON {}*/
-		if err!= nil{
-		fmt.Println("HTTP request failed with error")
-		} else{	
-			data, _:= ioutil.ReadAll(response.Body)
+		if err != nil {
+			fmt.Println("HTTP request failed with error")
```

Comment:
You should display the original error as well, so that the user can understand why it failed

Reply:
> You should display the original error as well, so that the user can understand why it failed

Yes I just changed it a while ago in my last update as I needed to see why it was failing